packages/js/components/src/utils.tsx

```diff
@@ -5,30 +5,69 @@ import { isValidElement, Fragment } from 'react';
 import { Slot, Fill } from '@wordpress/components';
 import { cloneElement, createElement } from '@wordpress/element';
 
+type ChildrenProps = {
+	order: number;
+};
+
 /**
- * Ordered fill item.
+ * Returns an object with the children and props that will be used by `cloneElement`. They will change depending on the
+ * type of children passed in.
  *
- * @param {Node}   children - Node children.
- * @param {number} order    - Node order.
- * @param {Array}  props    - Fill props.
- * @return {Node} Node.
+ * @param {Node}   children    - Node children.
+ * @param {number} order       - Node order.
+ * @param {Array}  props       - Fill props.
+ * @param {Object} injectProps - Props to inject.
+ * @return {Object} Object with the keys: children and props.
  */
-function createOrderedChildren< T = Fill.Props, S = Record< string, unknown > >(
+function getChildrenAndProps< T = Fill.Props, S = Record< string, unknown > >(
 	children: React.ReactNode,
 	order: number,
 	props: T,
 	injectProps?: S
 ) {
 	if ( typeof children === 'function' ) {
-		return cloneElement( children( { ...props, order, ...injectProps } ), {
-			order,
-			...injectProps,
-		} );
+		return {
+			children: children( { ...props, order, ...injectProps } ),
+			props: { order, ...injectProps },
+		};
 	} else if ( isValidElement( children ) ) {
-		return cloneElement( children, { ...props, order, ...injectProps } );
+		if ( typeof children?.type === 'function' ) {
```

Comment:
I was initially confused by this condition as I expected it to be the same as line 28: `if ( typeof children === 'function' ) {` but then realized it wasn't and it is to check if the child is a React element or not.

I think it would be worth adding a comment here to mention that this checks if it is a react element or a standard html element.

Reply:
Good idea @louwie17! I added that comment in the commit [2e7b0a3](https://github.com/woocommerce/woocommerce/pull/36707/commits/2e7b0a35461e96ed6fdaa943694c09f8dbbe16a2).

Could you give this PR another look?